plugins/inputs/tail/tail.go

```diff
@@ -223,6 +226,21 @@ func (t *Tail) tailNewFiles(fromBeginning bool) error {
 				// we're already tailing this file
 				continue
 			}
+
+			//Check whether tailing has started
+			flag = true
+			for e := l.Front(); e != nil; e = e.Next() {
+				if( e.Value == filepath) {
+					flag = false
+				}
+			}
+
+			if (flag == false) {
+				// we have already started tailing this file
+				continue
+			}
+			l.PushBack(filepath)
```

Comment:
There is a race condition between line 242 and 232. Two threads may add the same file. The both go through loop in line 232, don't find the file in the list, then proceed and both add the file to the list. Pleas find a root cause why the same file gets tailed by more than one thread.

Reply:
> There is a race condition between line 242 and 232. Two threads may add the same file. The both go through loop in line 232, don't find the file in the list, then proceed and both add the file to the list. Pleas find a root cause why the same file gets tailed by more than one thread.

It wont race, we are adding the entry to list only if that same entry is not already there in list. this is to make sure only one thread handles one named pipe.

Reply:
Imaging how two thread execute this code at the same time at the same lines. This is a classic race condition when you check if object is not added, then add it a few lines later.

Reply:
> Imaging how two thread execute this code at the same time at the same lines. This is a classic race condition when you check if object is not added, then add it a few lines later.

@dshnayder 2 threads wont reach this line with same filename